stingray/crossspectrum.py

```diff
@@ -445,16 +452,17 @@ def _make_segment_spectrum(self, lc1, lc2, segment_size):
         if lc1.tseg != lc2.tseg:
             raise ValueError("Lightcurves do not have same tseg.")
 
-        # number of bins per segment
-        nbins = int(segment_size/lc1.dt)
-        start_ind = 0
-        end_ind = nbins
+        if gti is None:
+            gti = cross_two_gtis(lc1.gti, lc2.gti)
```

Comment:
@dhuppenkothen I think I addressed all your points in this comment. Now, I first set the attribute, then use _make_segment_spectrum using that attribute :)